### PR TITLE
Fix web-worker-related bugs

### LINKED ISSUE #8519: amp-bind: Web worker bugs
1. https://ampbyexample.com/samples_templates/product/preview/ is broken with web-worker enabled due to invocation string typo in `bind-impl.js`.
2. Web worker entry point needs to call `initLogConstructor` due to nested log calls in `bind-evaluator.js`.

Examples that don't use dynamic elements like `amp-form` still work with web-worker enabled.

### PR DESCRIPTION
**Pull request by choumx**
Fixes #8519.

- Method string is `bind.removeBindings`, not `bind.removeBindingsWithExpressions`.
- Call `initLogConstructor` in entry point.
- Only invoke if operation is not a no-op and other misc. cleanup.

/to @kmh287

### CODE REVIEW

File: extensions/amp-bind/0.1/bind-expression.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ export class BindExpression {
     const skipConstraint = getMode().localDev && !getMode().test;
     if (size > maxSize && !skipConstraint) {
       throw new Error(`Expression size (${size}) exceeds max (${maxSize}). ` +
-          `Please reduce number of operands in: "${expressionString}"`);
+          `Please reduce number of operands.`);
```
Comment (kmh287) [inline extensions/amp-bind/0.1/bind-expression.js]: Why not include the expression?

File: extensions/amp-bind/0.1/bind-impl.js [hunk 2]
```diff
@@ -329,14 +331,17 @@ export class Bind {
...
-        return invokeWebWorker(this.win_,
-          'bind.removeBindingsWithExpressionStrings',
-          [deletedExpressions]);
-      } else {
-        this.evaluator_.removeBindingsWithExpressionStrings(deletedExpressions);
+      if (deletedExpressions.length > 0) {
+        if (this.workerExperimentEnabled_) {
+          dev().fine(TAG, `Asking worker to remove expressions...`);
+          return invokeWebWorker(
+              this.win_, 'bind.removeBindings', [deletedExpressions]);
```
Comment (kmh287) [inline extensions/amp-bind/0.1/bind-impl.js]: The method called on the evaluator is called `removeBindingsWithExpressionStrings`. Maybe this string should stay as it was and the string on the web worker needs to be updated?

File: extensions/amp-bind/0.1/bind-expression.js (hunk 1 above)
Comment (choumx) [inline extensions/amp-bind/0.1/bind-expression.js]: It's included in the wrapping user error in `bind-impl.js`.

File: extensions/amp-bind/0.1/bind-impl.js (hunk 2 above)
Comment (choumx) [inline extensions/amp-bind/0.1/bind-impl.js]: Done.

File: extensions/amp-bind/0.1/bind-impl.js [hunk 3]
```diff
@@ -116,6 +116,10 @@ export class Bind {
     /** @const @private {boolean} */
     this.workerExperimentEnabled_ = isExperimentOn(this.win_, 'web-worker');
 
+    if (!this.workerExperimentEnabled_) {
+      this.evaluator_ = new BindEvaluator();
```
Comment (kmh287) [inline extensions/amp-bind/0.1/bind-impl.js]: :+1:

**Pull request merged**